Source/Task/asyncQueueP.h

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+// This is the "private" api for an async queue.  It is an immutable definition used to help define
+// the implementation of a queue so that multiple implementations can co-habitate and share the same
+// flat public API.
+#pragma once
+
+enum class ApiId
+{
+    Identity,
+    AsyncQueueSection,
+    AsyncQueue
+};
+
+// Lightweight version of IUnknown
+struct IApi
+{
+    virtual uint32_t __stdcall AddRef() = 0;
+    virtual uint32_t __stdcall Release() = 0;
+    virtual HRESULT __stdcall QueryApi(_In_ ApiId id, _Out_ void** ptr) = 0;
```

Comment:
do we really need this?
QueryInterface is subtle and annoying to implement, do we really need runtime type information in this code?